### PR TITLE
Use groupcache.Contain to optimize HEAD request

### PR DESCRIPTION
**Pull request by anthonyjpratti**
This will improve performance of RPC method FindMissingBlobs.

Depends on https://github.com/twitter/groupcache/pull/11.

### CODE REVIEW

**Comment by dgdonut** [comment]
Can you link include a link to the groupcache pr this depends on?

File: snapshot/bundlestore/server.go [hunk 1]
```diff
@@ -43,11 +43,12 @@ func MakeServer(s store.Store, ttl *store.TTLConfig, stat stats.StatsReceiver, g
 // Implements http.Handler interface
 func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	s.storeConfig.Stat.Counter(stats.BundlestoreRequestCounter).Inc(1)
+	//TODO(apratti): write ttl in headers in each of these
```
Comment (dgdonut) [inline snapshot/bundlestore/server.go]: I don't think this is part of the contract this server's API guarantees

File: snapshot/bundlestore/http_server.go [hunk 2]
```diff
@@ -79,6 +79,28 @@ func (s *httpServer) HandleUpload(w http.ResponseWriter, req *http.Request) {
...
+		return
+	}
+	ok, err := s.storeConfig.Store.Exists(bundleName)
+	if err != nil || !ok {
+		log.Infof("Check err: %v --> StatusNotFound (from %v)", err, req.RemoteAddr)
+		http.NotFound(w, req)
+		s.storeConfig.Stat.Counter(stats.BundlestoreCheckErrCounter).Inc(1)
+		return
+	}
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```
Comment (dgdonut) [inline snapshot/bundlestore/http_server.go:100]: why set this header?

File: snapshot/store/groupcache_store.go [hunk 3]
```diff
@@ -65,6 +65,17 @@ func MakeGroupcacheStore(underlying Store, cfg *GroupcacheConfig, ttlc *TTLConfi
 			}
 			return ttl, dest.SetBytes(data)
 		}),
+		groupcache.ContainerFunc(func(ctx groupcache.Context, bundleName string) (bool, error) {
+			log.Info("Not cached, try to check bundle existence and populate cache metadata: ", bundleName)
```
Comment (dgdonut) [inline snapshot/store/groupcache_store.go]: log msg - doesn't try to populate cache metadata

File: snapshot/bundlestore/http_server.go (hunk 2 above)
Comment (anthonyjpratti) [reply snapshot/bundlestore/http_server.go:100]: To match the GET response, the http lib adds this automatically for GETs.

File: snapshot/bundlestore/server.go (hunk 1 above)
Comment (anthonyjpratti) [reply snapshot/bundlestore/server.go]: Will remove

File: snapshot/store/groupcache_store.go (hunk 3 above)
Comment (anthonyjpratti) [reply snapshot/store/groupcache_store.go]: Will fix

**Review by dgdonut — APPROVED** [review]

**Pull request merged**